microstream/src/main/java/io/micronaut/microstream/cache/MicrostreamSyncCache.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.microstream.cache;
+
+import io.micronaut.cache.SyncCache;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.type.Argument;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.cache.Cache;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * A {@link SyncCache} implementation that uses a Microstream Cache instance.
+ *
+ * @param <K> the key type
+ * @param <V> the value type
+ * @since 1.0.0
+ * @author Tim Yates
+ */
+public class MicrostreamSyncCache<K, V> implements SyncCache<Cache<K, V>>, AutoCloseable {
```

Comment:
Is this not essentially the same as https://github.com/micronaut-projects/micronaut-cache/blob/master/cache-core/src/main/java/io/micronaut/cache/jcache/JCacheSyncCache.java ?

Another option could be https://github.com/micronaut-projects/micronaut-cache/blob/master/cache-core/src/main/java/io/micronaut/cache/AbstractMapBasedSyncCache.java

Reply:
Another option could be to introduce an `AbstractJCacheBasedSyncCache` in micronaut-cache

Reply:
I think @alvarosanchez is correct, we should do `JCacheSyncCache` public.

Reply:
https://github.com/micronaut-projects/micronaut-cache/pull/459